File: Python/_129_Sum_Root_to_Leaf_Numbers/solution.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def sumNumbers(self, root):
        """
        :type root: TreeNode
        :rtype: int
        """
        return self.dfs(root, 0)
        
    def dfs(self, root, res):
        if not root:
            return 0
        res = 10 * res + root.val
        if not root.left and not root.right:
            return res
        return self.dfs(root.left, res) + self.dfs(root.right, res)

File: Java/_129_Sum_Root_to_Leaf_Numbers/Solution.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */
class Solution {
    public int sumNumbers(TreeNode root) {
        return dfs(root, 0);
    }
    
    public int dfs(TreeNode root, int res) {
        if (root == null)
            return 0;
        res = 10 * res + root.val;
        if (root.left == null && root.right == null)
            return res;
        return dfs(root.left, res) + dfs(root.right, res);
    }
}